src/alarms/contacts/NewContact.js

```diff
@@ -31,55 +31,82 @@ class NewContact extends Component {
-        headers: { "Content-Type": "application/json" },
-        body: JSON.stringify({
-          first_name: firstName,
-          last_name: lastName,
-          email: email,
-          phone_number: phoneNumber,
-          organisation: organisation.uuid
-        })
-      })
-        .then(response => {
-          if (response.status === 201) {
-            return response.json();
-          } else if (response.status === 403) {
-            alert("Not authorized");
-            return response.json();
-          } else {
-            alert("An error occurred while trying to create a new contact");
-            return response.json();
-          }
-        })
-        .then(data => {
-          history.push("/alarms/contacts");
-        });
-    } else {
-      alert("Please provide valid values in all fields!");
+    const firstNameTranslation = this.props.intl.formatMessage({ id: "contacts_app.first_name_3" });
+    const lastNameTranslation = this.props.intl.formatMessage({ id: "contacts_app.last_name_3" });
+    const emailTranslation = this.props.intl.formatMessage({ id: "contacts_app.email_address_3" });
+    // const phoneNumberTranslation = this.props.intl.formatMessage({ id: "contacts_app.phone_number_3" });
+    const fieldsDidnotValidateTranslation =  this.props.intl.formatMessage({ id: "contacts_app.following_fields_didnot_validate" }) + ' \n '; 
+
+    let errorMessage = fieldsDidnotValidateTranslation;
+    let showErrorMessage = false;
+
+    if ( firstName.length == 0 ) {
+      errorMessage += "- " + firstNameTranslation;
+      showErrorMessage = true;
+    }
+    if (lastName.length == 0 ) {
+      errorMessage += "- " + lastNameTranslation;
+      showErrorMessage = true;
+    }
+    if (!this.validateEmail(email)) { // && phoneNumber.length == 0  
+      errorMessage += "- " + emailTranslation; // + " OR " + phoneNumberTranslation;
+      showErrorMessage = true;
+    }
+
+    if (showErrorMessage) {
+      alert(errorMessage);
+      return;
     }
+
+    this.setState({
+      submitActive: true
+    });
+    fetch("/api/v3/contacts/", {
+      credentials: "same-origin",
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        first_name: firstName,
+        last_name: lastName,
+        email: email,
+        phone_number: phoneNumber,
+        organisation: organisation.uuid
+      })
+    })
+    .then(response => {
+      if (response.status === 201) {
+        return response.json();
+      } else if (response.status === 403) {
+        alert("Not authorized");
+        return response.json();
+      } else {
+        alert("An error occurred while trying to create a new contact");
+        return response.json();
+      }
+    })
+    .then(data => {
+      history.push("/alarms/contacts");
+    });
   }
   render() {
     return (
       <div className={gridStyles.Container}>
-        <div className={`${gridStyles.Row}`}>
+        <div
```

Comment:
Why do you need this field here with "display:none"? I don't see anywhere in the code you want to display this field at later stages so it is quite redundant in my opinion. Maybe you have another intention for it :)

Reply:
Also true. It will never be visible for the user. It is for the translation engine so that it is able to extract the string. I will add a comment in the code.